src/boot/efi/linux.c

```diff
@@ -128,7 +128,7 @@ EFI_STATUS linux_exec(EFI_HANDLE *image,
                 if (EFI_ERROR(err))
                         return err;
                 CopyMem((VOID *)(UINTN)addr, cmdline, cmdline_len);
-                ((CHAR8 *)addr)[cmdline_len] = 0;
+                ((CHAR8 *)(UINTN)addr)[cmdline_len] = 0;
```

Comment:
So, `addr` is a `EFI_PHYSICAL_ADDRESS`, which is 64 bits. This casts is to UINTN, which is 32 bits on 32 bit. For this to work at least `BS->AllocatePages` must return an address that fits in the right half of `addr`. Do we know this?

Reply:
Well, it's a 32bit system, hence only 32bit can be addressed, i.e. if an address stored in an EFI_PHYSICAL_ADDRESS is valid at all, then it will be in the 32bit range, even if the type allows larger values in theory. 

I mean, we could certainly add a check that `addr` is in the 32bit range, but that would be second guessing the API as in "did the API really return a valid address to us?", and such a test of course would be very incomplete, because an address can be invalid in many other ways too, besides being outside of the addressable range...

Hence, yes, we *do* know it's not going to be outside of the addressable range.